### ISSUE TITLE
New feature: webpack bundle stats plugin

### ISSUE THREAD

**Issue by jakoblind** [jakoblind/webpack-autoconf#71]
There are many stats plugin such as [source-maps-explorer](https://github.com/danvk/source-map-explorer) and [webpack-bundle-analzyer](https://github.com/webpack-contrib/webpack-bundle-analyzer) and others. which one is most "best practice"? We should pick that and add it as a feature.

This should be under the new feature group "Webpack plugins" as also discussed in https://github.com/jakoblind/webpack-autoconf/issues/64

**Comment by atilkan** [comment]
webpack-bundle-analzyer is more visual and has the what the other has.

**Comment by jacekk** [comment]
I also recommend https://www.npmjs.com/package/bundlesize as it can be easily triggered in CI and fail a job when some assets grow too much, and which could/should be optimized.

**Comment by linus345** [comment]
I can give this a try, [webpack-bundle-analyzer](https://github.com/webpack-contrib/webpack-bundle-analyzer) looks good to me. What would you like the feature to be called? `Bundle stats` or `Webpack Bundle Analyzer` maybe?

**Comment by linus345** [comment]
webpack-bundle-analyzer gives a [option](https://github.com/webpack-contrib/webpack-bundle-analyzer#options-for-plugin) called `analyzerMode` which lets you configure if you want the plugin to start a server to show the report or generate a static HTML file of the bundle report. I think generating a static HTML file would be the best option. What do you think?

**Comment by jakoblind** [comment]
I think we can support many of these tools. We could add a new group called "Analyzers" or something, and then add "Webpack Bundle Analyzer" to that group, and possibly more later. https://www.npmjs.com/package/bundlesize like @jacekk suggested could be the next candidate maybe.

Regarding the `analyzerMode` option, we should use the most best-practice way of using it, whatever that might be. I'm not sure what's the best thing here. Do you think it should be run every time we run `npm run build-prod` ? Then I agree that generating static report is better. Or we could create a separate NPM script so that we could run `npm run stats`. I think personally that would be cleaner solution. If we would create a separate NPM script it could maybe run the following:

```
webpack --profile --json > stats.json
webpack-bundle-analyzer bundle/output/path/stats.json
```

**Comment by linus345** [comment]
I agree that creating a separate NPM script would be a good solution. Should we decide on that? @jacekk and @atilkan do you have any other suggestions?

**Comment by jakoblind** [comment]
Yes let's decide on that! and maybe put it under "webpack plugins" section until we have more stats plugins implemented.

**Comment by linus345** [comment]
There seems to be an [issue](https://github.com/webpack-contrib/webpack-bundle-analyzer/issues/47) on Windows with the `webpack --profile --json > stats.json` command that generates the json file. If Windows is a requirement another option to generate the json file could be to [use it as a plugin](https://github.com/webpack-contrib/webpack-bundle-analyzer#usage-as-a-plugin) and set the option `analyzerMode` to `disabled` and `generateStatsFile` to `true` which will create the json file that could be used with an NPM script that would run `webpack-bundle-analyzer bundle/output/path/stats.json` to see the status report.

**Comment by jakoblind** [comment]
I think Windows is a requirement. 

Maybe we should go back to what you proposed initially. Just add the config to `webpack.config.js` with `analyzerMode` set to generate static files. Something like this:

```
const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;

module.exports = {
  plugins: [
    new BundleAnalyzerPlugin({static: "static"})
  ]
}
```

It's simple and it makes sense. Also it's proposed at the top of the readme of `webpack-bundle-analyzer`
